2_run/src/run_glm3.R

```diff
@@ -133,20 +131,41 @@ run_glm3_model <- function(sim_dir, nml_objs, model_config, burn_in, burn_out, e
           glm_time <- system.time({glm_code <- GLM3r::run_glm(sim_lake_dir, verbose = FALSE)})[['elapsed']]
           # Pull out the final date from the output to
           # check against requested simulation end date
-          output_dates <- glmtools::get_temp(nc_filepath) %>%
+          # use 1D variable 'evap' rather than 2D 'temp' to check output dates because
+          # 1) less expensive to pull and 2) defaults cause an error with glmtools::get_temp()
+          # when the lake is too shallow.
+          output_dates <- glmtools::get_var(nc_filepath, var_name = 'evap') %>%
             mutate(date = format(as.Date(lubridate::floor_date(DateTime, 'days')),"%Y-%m-%d")) %>%
             pull(date)
           max_output_date <- max(output_dates)
         },
         until=function(val, cnd) glm_code == 0 & max_output_date==sim_stop,
         max_tries = 5)
-
+      
       # make sure glm did succeed  
       if(glm_code != 0 | max_output_date!=sim_stop) stop()
       
       # extract output
       export_fl <- sprintf(export_fl_template, site_id, gcm, time_period)
-      extract_glm_output(nc_filepath, nml_obj, export_fl)
+      extraction_error <- tryCatch(
+        {
+          extract_glm_output(nc_filepath, nml_obj, export_fl)
+          extraction_error <- NA
+        },
+        error = function(e) {
+          if (grepl(e$message, 'need at least two non-NA values to interpolate')) {
+            extraction_error <- 'NA values'
+          } else {
+            extraction_error <- e$message
+          }
+          return(extraction_error)
+        }
+      )
+      # If extraction error is NOT NA, trigger error
+      if (!is.na(extraction_error)) stop()
+      
+      # If output was extracted, delete simulation directory
+      unlink(sim_lake_dir, recursive = TRUE)
```

Comment:
Just to clarify, you don't want to delete the ones that failed so that you can explore, correct?